task_02/shamil_latypov/graph.cpp

```diff
@@ -0,0 +1,127 @@
+#include <iostream>
+#include <stdlib.h>
+#include <map>
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+
+struct edge {
+    int id = 0;
+    //std::string name = "";
+};
+
+struct vertex {
+    int id = 0;
+    
+    friend bool operator< (vertex a, vertex b) {
+        return a.id < b.id;
+    }
+    
+    friend bool operator== (vertex a, vertex b) {
+        return a.id == b.id;
+    }
+
+    //std::string name = "";
+};
+
+struct graph {
+    struct vertex v1;
+    struct edge ed;
+    struct vertex v2;
+
+    friend bool operator< (graph a, graph b) {
+        if (a.v1 == b.v1) return a.v2 < b.v2;
+        else return a.v1 < b.v1;
+    }
+};
+
+/*
+void set_param (num, change_param) {
+    struct[num].param = change_param;
+}
+*/
+
+void get_graph(std::vector <struct graph> *all_graph, std::pair <vertex, bool> vert_mas[], int NOEdge, int NOVertex)
```

Comment:
1. `struct graph`, `vert_mas[]` это синтаксис `C`, а не `C++`, давай не будем таким баловаться :)
У нас, все-таки практикум по `C++`
2. Зачем ты передаешь `all_graph` как поинтер?
3. Что за `NOEdge` и `NOVertex`? Это значит `number of vertices`? Тогда лучше переименовать их, потому что:
    - Во-первых, переменные не называются с большой буквы
    - Во-вторых, если в названии типа в начале идет несколько заглавных букв, то это обычно какая-то аббревиатура. 

```suggestion
void get_graph(const std::vector<Graph>& all_graph, std::vector<std::pair<vertex, bool>> vert_mas, int edge_count, int vertices_count)
```